analysers/analyser_osmosis_railway_angle.py

```diff
@@ -0,0 +1,111 @@
+# However, this one measures angles and not radiuses
+# It most likely indicates a node that was accidentaly dragged
+# While Analyser_Osmosis_Way_Approximate indicates that the way was drawn
+# with too few nodes
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Osmosis import Analyser_Osmosis
+
+sql10 = """
+WITH
+    points AS (
+        SELECT
+            id,
+            tags->'{tag}' AS type,
+            (ST_DumpPoints(linestring)).geom AS geom4326,
+            (ST_DumpPoints(ST_Transform(linestring, {srid}))).geom AS geom,
+            (ST_DumpPoints(ST_Transform(linestring, {srid}))).path[1] AS index
+        FROM
+            {table_prefix}ways as ways
+        WHERE
+            tags != ''::hstore AND
+            tags?'{tag}' AND tags->'{tag}' IN ('{tag_values}') AND
+            ST_NPoints(linestring) >= 3
+    ),
+    angles AS (
+        SELECT
+            id, index, type,
+            lead(geom4326) OVER (PARTITION BY id ORDER BY index) AS geom4326,
+            lead(geom) OVER (PARTITION BY id ORDER BY index) AS geom,
+            degrees(
+                ST_Angle(geom,
+                     lead(geom, 1) OVER (PARTITION BY id ORDER BY index),
+                     lead(geom, 2) OVER (PARTITION BY id ORDER BY index)
+                )
+            ) angle,
+            ST_Distance(
+                geom,
+                lead(geom, 1) OVER (PARTITION BY id ORDER BY index)
+            ) AS length_1,
+            ST_Distance(
+                lead(geom, 1) OVER (PARTITION BY id ORDER BY index),
+                lead(geom, 2) OVER (PARTITION BY id ORDER BY index)
+            ) AS length_2
+        FROM
+            points
+    )
+
+    SELECT
+        id,
+        ST_AsText(geom4326),
+        angle,
+        type,
+        index
+    FROM
+        angles
+    WHERE
+        (angle < 150 OR angle > 210)
+        AND length_1 < 100
+        AND length_2 < 100
+"""
+
+class Analyser_Osmosis_Rail_Suspicious_Angle(Analyser_Osmosis):
+
+    def __init__(self, config, logger = None):
+        Analyser_Osmosis.__init__(self, config, logger)
+
+        self.classs_change[10] = self.def_class(item = 1190, level = 2, tags = ['geom', 'railway', 'fix:imagery'],
+            title = T_('Suspicious angle in railway'),
+            detail = T_(
+'''Sharp angles on a railway are suspicious. Maybe a node was accidentaly dragged'''),
+            fix = T_(
+'''After checking orthophotos, add nodes or move existing nodes.'''),
+            trap = T_(
+'''On service ways, train stations, train workshops that may be either a
+false positive'''),
+            example = T_(
+'''![](https://wiki.openstreetmap.org/w/images/9/9d/Osmose-eg-error-1190.png)
+
+`railway=rail` crudely drawn.'''))
+
+        self.tags = ("rail", "lightrail")
+        self.callback10 = lambda res: {"class":10, "subclass":res[2], "data":[self.way_full, self.positionAsText], "text": T_("railway with suspicious angle {0}°", res[1])}
```

Comment:
Analyser way_approximate uses the same class and item id. To allow users to filter, I would use class `11` here (and thus also in line 90)

---

Additionally, I think the angle is in `res[2]`?
The current message reads: `railway with suspicious angle POINT(81.36814960000001 50.936915400000004)°` with `subclass="352.6731599505589"` (the angle).
(I think subclass must be an integer? Maybe this is where you wanted to use `index` (=`res[4]`)?

`res[3]` (the value of tag `railway=*`) is currently not used.